File: Problems/1603. (E) Design Parking System/Python3/solution.py
class ParkingSystem:
  def __init__(self, big: int, medium: int, small: int):
    self._l = big
    self._m = medium
    self._s = small

  def addCar(self, carType: int) -> bool:
    match carType:
      case 1:
        if self._l:
          self._l -= 1
          return True
      case 2:
        if self._m:
          self._m -= 1
          return True
      case 3:
        if self._s:
          self._s -= 1
          return True
      case _:
        return False
    return False

File: Problems/1603. (E) Design Parking System/Java/solution.java
class ParkingSystem {
  private int l;
  private int m;
  private int s;

  public ParkingSystem(int big, int medium, int small) {
    l = big;
    m = medium;
    s = small;
  }
  
  public boolean addCar(int carType) {
    switch (carType) {
      case 1:
        if (this.l > 0) {
          this.l--;
          return true;
        } break;
      case 2:
        if (this.m > 0) {
          this.m--;
          return true;
        } break;
      case 3:
        if (this.s > 0) {
          this.s--;
          return true;
        }
    }
    return false;
  }
}
